### ISSUE TITLE
2FA QRCode image is not generated for servers located in Mainland China

### ISSUE THREAD

**Issue by MarcSamD** [nilsteampassnet/TeamPass#3645]
### Steps to reproduce
1. Teampass server located behind the GFW (or just block access to chart.googleapis.com in your server to simulate)
2. Enable "Enable Google 2-step Verification"
3. Login as a user

### Expected behaviour
QRCode should be displayed

### Actual behaviour
QRCode is not displayed:
![image](https://user-images.githubusercontent.com/70358310/231379470-ed17298e-3909-4bdb-a785-5de89833f2bc.png)

### Workaround (dirty)
Edit the file GoogleQRCodeProvider.php to replace the function getUrl as below:
```
    public function getUrl($qrtext, $size)
    {
        return 'https://api.qrserver.com/v1/create-qr-code/'
            . '?size=' . $size . 'x' . $size
            . '&ecc=' . strtoupper($this->errorcorrectionlevel)
            . '&margin=' . $this->margin
            . '&data=' . rawurlencode($qrtext);
    }

```

### Solution (expected)
There are several providers already existing in the Authentication libraries (especially QRServerProvider.php which is for api.qrserver.com), but there is no way to select them in the front-end (only Duo Security and Google are available).

### Partialy related issues
https://github.com/nilsteampassnet/TeamPass/issues/3287
https://github.com/nilsteampassnet/TeamPass/issues/3477

**Comment by tbalbers** [comment]
I agree with MarcSamD, and I'd like to add that having an external dependency on a security tool like this is a bad idea.
Ideally, the functionality to generate a QR for syncing a TOTP app should be integrated in TeamPass. If you take a look at SysPass, they have this feature integrated.

**Comment by nilsteampassnet** [comment]
Closed automatically due to inactivity (no response for over 1 year).
